### PR TITLE
feat(medusa,core-flows): update cart adjustments on item updates

### PR DESCRIPTION
**Pull request by riqwan**

### CODE REVIEW

File: packages/core-flows/src/definition/cart/steps/refresh-cart-promotions.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    | PromotionActions.REPLACE
+}
+
+export const refreshCartPromotionsStepId = "refresh-cart-promotions"
+export const refreshCartPromotionsStep = createStep(
+  refreshCartPromotionsStepId,
+  async (data: StepInput, { container }) => {
+    const { promo_codes = [], id, action = PromotionActions.ADD } = data
+
+    await updateCartPromotionsWorkflow(container).run({
```
Comment (olivermrbl) [inline packages/core-flows/src/definition/cart/steps/refresh-cart-promotions.ts:20]: love this! 🎆

File: packages/core-flows/src/definition/cart/workflows/update-cart.ts [hunk 2]
```diff
@@ -79,34 +62,24 @@ export const updateCartWorkflow = createWorkflow(
...
-      removeShippingMethodAdjustmentsStep({
-        shippingMethodAdjustmentIdsToRemove,
-      })
-    )
-
-    parallelize(
-      createLineItemAdjustmentsStep({ lineItemAdjustmentsToCreate }),
-      createShippingMethodAdjustmentsStep({ shippingMethodAdjustmentsToCreate })
-    )
+    const retrieveCartInput = {
```
Comment (olivermrbl) [inline packages/core-flows/src/definition/cart/workflows/update-cart.ts:71]: question: Can we do this? I didn't know that. Thought everything had to be either a step or transformer.

Comment (riqwan) [reply packages/core-flows/src/definition/cart/workflows/update-cart.ts:71]: We changed it recently to allow plain objects as well. You only need the transformer if its an unresolved object.

Comment (olivermrbl) [reply packages/core-flows/src/definition/cart/workflows/update-cart.ts:71]: That's great

Comment (carlos-r-l-rodrigues) [reply packages/core-flows/src/definition/cart/workflows/update-cart.ts:71]: one can't keep up with so many improvements 😆

Comment (olivermrbl) [reply packages/core-flows/src/definition/cart/workflows/update-cart.ts:71]: Haha, let's keep it that way 🏎️

**Pull request merged**